src/hdr-config.cpp

```diff
@@ -232,18 +232,13 @@ namespace librealsense
             {
                 hwmon_response response;
                 auto res = _hwm.send( cmd, &response );  // avoid the throw
-                switch( response )
+                if (response != _hwm.hwmon_response_handler->hwmon_Success()) // If no subpreset is streaming, the firmware returns "NO_DATA_TO_RETURN" error
```

Comment:
This is not exactly equivalent to the former logic. Why was this changed?

Reply:
I think it's becasue now we get the success value from a virtual function. I'm getting an error when trying to use `_hwm.hwmon_response_handler->hwmon_Success()` as a 'case' value (`'this' can not be used in a constant expression`)

Reply:
Yes, but now you don't have any reference to `hwmon_response::hwm_NoDataToReturn`, is it not needed?
You can replace `switch` with some `else if` clauses, but here you have changed the logic. Just make sure its intended and works as expected.

Reply:
oh I see, thanks, I will add `no_data_to_return_opcode` as a member to this class, like we did on alternating_emitter_option